### ISSUE TITLE
Persistent games

### ISSUE THREAD

**Issue by ctoth** [soundmud/soundrts#52]
Suggesting the idea of allowing people to both join long-running games and re-join games. This will require players being assigned a uuid by the server, so when they reconnect, it can be verified they should be able to control whatever games.
This would possibly allow multiday, or even longer games, with units patrolling ones empire and keeping it safe while offline, and logging in to make moves when available.
Much more of a longterm pie-in-the-sky goal, but something worth thinking about as we shape the upcoming changes in architecture.

**Comment by soundmud** [comment]
Yes, this would turn the RTS into a MMO RTS, which isn't necessarily a bad thing, just a very different game. I have experienced a little with persistence in SoundMud Redux, and I have noticed that persistent worlds "remember" design errors and balance problems, and this is an additional problem.
In this case again, such a game isn't necessarily harder to code, but it's something very different from the current game. So maybe restarting a completely new game from scratch would be much easier.

**Comment by sanslash332** [comment]
MMO rts?

What are you thinking? Some like oGame, but instead play in browser, that it have it’s own client?

 

The idea isn’t bad, but this is for a complete different game. You have to check it frequently, you play against all opponent that are playing the game in all moments, and the real time of the game is not calculate in seconds, but in howrs, or days.

 

This… is so far the classic concept of a rts game.

**Comment by soundmud** [comment]
Yes, I was thinking about something like OGame or Travian, but with an "immersive" client and a "second by second" detailed simulation. The number of players would be relatively small (for example 20 players) or the server wouldn't be able to do the simulation properly. I'm thinking about a mix of "long" travels to the enemy and "short" real time encounters. A siege might take a long time too. Some realistic aspects might have to be "balanced" for fun, like the time scale (travel, combat, building time). A single player version would allow the time to accelerate when "nothing" is happening.

**Comment by sanslash332** [comment]
Yep, the idea is simply good :P

But, for this you should start a new project, instead of modify the current SOUNDRTS, because currently soundrts aren’t finished yet, it can do a lot of more things.

**Comment by soundmud** [comment]
About saving and resuming online games, I might have a simple solution. I have added an issue about this: https://github.com/soundmud/soundrts/issues/54

**Comment by tuohai** [comment]
Yes, if you want to do a good job, it is best to re write a soundrts themselves, adding content themselves, such as resources, race also do not need too much, 4 to 3 can be

**Comment by soundmud** [comment]
Saving and resuming games shall happen soon, I hope.
Allowing players to join an already started game (issue #94) is a short term goal too.
A completely automated persistent server seems a bit too far right now, but with these future features a manually managed persistent game could be possible.

**Comment by tuohai** [comment]
good.
but， Once the number of units increases or map large, the game will Hysteresis behind. Is it really impossible to solve this problem?

**Comment by soundmud** [comment]
I have added issue #98 about speed.

**Comment by tuohai** [comment]
That's great. It's definitely a big progress as long as it's solved.
